src/main/java/io/blueocean/rest/pipeline/editor/PipelineStepMetadataService.java

```diff
@@ -0,0 +1,214 @@
+package io.blueocean.rest.pipeline.editor;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.workflow.cps.Snippetizer;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.Stapler;
+import org.kohsuke.stapler.WebMethod;
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+import org.kohsuke.stapler.verb.GET;
+import org.springframework.core.LocalVariableTableParameterNameDiscoverer;
+import org.springframework.core.ParameterNameDiscoverer;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import io.jenkins.blueocean.commons.stapler.TreeResponse;
+import io.jenkins.blueocean.rest.ApiRoutable;
+import jenkins.model.Jenkins;
+
+/**
+ * This provides and Blueocean REST API endpoint to obtain pipeline step metadata.
+ */
+@Extension
+public class PipelineStepMetadataService implements ApiRoutable {
```

Comment:
@kzantow https://issues.jenkins-ci.org/browse/JENKINS-38584 talks about serving this editor in context of an organization, that is /rest/organizations/:id/editor/steps. ApiRoutable is going to result in to serving it from /rest. 

Having said that, at present /organizations/:id/ doesn't have extension point to let plugins expose their object model as sub path to it. Its a WIP in https://github.com/jenkinsci/blueocean-plugin/pull/593. I should probably do this as separate PR as it might make it easy for you to expose this REST api at organization context.